src/fs.rs

```diff
@@ -276,6 +300,64 @@ fn handle_open_err<T>(open_result: std::io::Result<T>, host_path: &Path) -> T {
     }
 }
 
+/// Like [std::fs::File] but for the guest filesystem.
+#[derive(Debug)]
+pub enum File {
+    HostFile(std::fs::File),
+    IpaBundleFile(IpaFile),
+}
+
+impl File {
+    fn from_host_file(file: std::fs::File) -> File {
+        File::HostFile(file)
+    }
+
+    fn from_ipa_file(file: IpaFile) -> File {
+        File::IpaBundleFile(file)
+    }
+
+    pub fn sync_all(&self) -> std::io::Result<()> {
+        match self {
+            File::HostFile(file) => file.sync_all(),
+            File::IpaBundleFile(_) => Ok(()),
+        }
+    }
+}
+
+impl Read for File {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        match self {
+            File::HostFile(file) => file.read(buf),
+            File::IpaBundleFile(file) => file.read(buf),
+        }
+    }
+}
+
+impl Write for File {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        match self {
+            File::HostFile(file) => file.write(buf),
+            File::IpaBundleFile(file) => file.write(buf),
```

Comment:
I think we'll always want to treat the IPA file as read-only. I see you have a `Write` implementation in `src/fs/bundle.rs`, but maybe you could remove it and just panic here instead.